pal/pacer/pacer-web.ts

```diff
@@ -96,32 +97,41 @@ export class Pacer {
         this._stHandle = this._stTime(updateCallback);
 
         this._isPlaying = true;
+        this._frameCount = 0;
     }
 
     stop (): void {
         if (!this._isPlaying) return;
         this._ctTime(this._stHandle);
         this._stHandle = 0;
         this._isPlaying = false;
+        this._frameCount = 0;
     }
 
     _handleRAF = () => {
-        if (performance.now() - this._start < this._delay) {
+        const elapseTime = performance.now() - this._startTime;
+        const elapseFrame = Math.floor(elapseTime / this._frameTime);
+        if (elapseFrame < this._frameCount) {
             this._rAF.call(window, this._handleRAF);
-        } else if (this._callback) {
-            this._callback();
+        } else {
+            this._frameCount++;
+            if (this._callback) {
+                this._callback();
+            }
+        }
+        if (performance.now() - this._startTime > FRAME_RESET_TIME) {
```

Comment:
Why need to reset?

Reply:
The previous method was to calculate the delay frame by frame and use raf to run each delay frame. For the problems caused, see issue: https://github.com/cocos/3d-tasks/issues/16538#issuecomment-1575960596. Now it is to accumulate the delay for a period of time, tick one frame when it is longer than `_frameTime`, otherwise just delay to next frame.
Because `_frameCount` cannot be accumulated infinitely, it needs to be reset at a certain event point.

Reply:
Got it. How you determine the value of `FRAME_RESET_TIME `? It seems a magic number. Have you tested with different fps setting?

Reply:
> 2000 represents do reset each 2 seconds, that is, a reset is performed every 120 ticks on a 60hz screen. I'm not sure whether 2000 is the most suitable value. However, there is no problem with running, so it may be an appropriate value

In my opinion, as long as the value is not too small (does not cause calculation errors), nor too large (does not cause overflow).

> Have you tested with different fps setting?

Yes. I've different fps on mac and windows. This value seems run well.